addons/l10n_dk_oioubl/models/account_edi_xml_oioubl_201.py

```diff
@@ -0,0 +1,285 @@
+    'NO': 'NO:VAT',
+    'PL': 'PL:VAT',
+    'PT': 'PT:VAT',
+    'RO': 'RO:VAT',
+    'RS': 'RS:VAT',
+    'SE': 'SE:VAT',
+    'SI': 'SI:VAT',
+    'SK': 'SK:VAT',
+    'SM': 'SM:VAT',
+    'TR': 'TR:VAT',
+    'VA': 'VA:VAT',
+}
+
+
+def format_vat_number(partner):
+    vat = partner.vat.replace(' ', '')
+    if vat[:2].isnumeric():
+        vat = partner.country_code.upper() + vat
+    return vat
+
+
+class AccountEdiXmlOIOUBL201(models.AbstractModel):
+    _name = "account.edi.xml.oioubl_201"
+    _inherit = 'account.edi.xml.ubl_20'
+    _description = "OIOUBL 2.01"
+
+    # Data validation Schematron available at the following URL:
+    # https://rep.erst.dk/git/openebusiness/common/-/tree/master/resources/Schematrons/OIOUBL
+
+    # -------------------------------------------------------------------------
+    # EXPORT
+    # -------------------------------------------------------------------------
+
+    def _export_invoice_filename(self, invoice):
+        return f"{invoice.name.replace('/', '_')}_oioubl_201.xml"
+
+    def _export_invoice_ecosio_schematrons(self):
+        return {
+            'invoice': 'org.oasis-open:invoice:2.0',
+            'credit_note': 'org.oasis-open:creditnote:2.0',
+        }
+
+    def _export_invoice_vals(self, invoice):
+        vals = super()._export_invoice_vals(invoice)
+        vals.update({
+            'AddressType_template': 'l10n_dk_oioubl.oioubl_201_AddressType',
+            'PartyType_template': 'l10n_dk_oioubl.oioubl_201_PartyType',
+            'TaxCategoryType_template': 'l10n_dk_oioubl.oioubl_20_TaxCategoryType',
+        })
+        vals['vals'].update({
+            'customization_id': 'OIOUBL-2.01',
+            'AddressType_template': 'l10n_dk_oioubl.oioubl_201_AddressType',
+            # ProfileID is the property that define which documents the company can send and receive
+            # 'Procurement-BilSim-1.0' is the simplest one: invoice and bill
+            # https://www.oioubl.info/documents/en/en/Guidelines/OIOUBL_GUIDE_PROFILES.pdf
+            'profile_id': 'Procurement-BilSim-1.0',
+            'profile_id_attrs': {
+                'schemeID': 'urn:oioubl:id:profileid-1.6',
+                'schemeAgencyID': DANISH_NATIONAL_IT_AND_TELECOM_AGENCY_ID,
+            }
+        })
+        vals['vals'].setdefault('invoice_type_code_attrs', {}).update({
+            'listID': 'urn:oioubl:codelist:invoicetypecode-1.2',
+            'listAgencyID': DANISH_NATIONAL_IT_AND_TELECOM_AGENCY_ID,
+        })
+
+        return vals
+
+    def _get_partner_party_vals(self, partner, role):
+        vals = super()._get_partner_party_vals(partner, role)
+        vat = format_vat_number(partner)
+        vals.update({
+            # list of possible endpointID available at
+            # https://www.oioubl.info/documents/en/en/Guidelines/OIOUBL_GUIDE_ENDPOINT.pdf
+            'endpoint_id': vat,
+            'endpoint_id_attrs': {'schemeID': ENDPOINTID_SCHEMEID.get(vat[:2], 'DK:CVR')},
```

Comment:
So you'd be filling the vat while marking it as siret for France ?

Reply:
![image](https://github.com/odoo/odoo/assets/52834046/bf21865b-414d-42c7-ba1e-ac4d716cc008)


Yes and it seems to work as I have made this test for it: `test_export_invoice_two_line_schematron_foreign_partner_fr`

I validated using schematron, see this branch: 
https://github.com/odoo-dev/odoo/blob/59f32bf7936fd491df8999051f5768c5b1706699/addons/l10n_account_edi_ubl_cii_tests/tests/test_xml_oioubl_dk.py#L223-L273

Reply:
use company_registry when fr partner
+ add validation constraint --> company_registry has to be set when the partner is french